### DISCUSSION TITLE
martinize2 -p backbone create NO restraints

### DISCUSSION THREAD

**Discussion by Toolxx** [marrink-lab/vermouth-martinize#618] [Q&A]
I am following the tutorial "https://cgmartini.nl/docs/tutorials/Martini3/ProteinsII/" and "https://cgmartini.nl/docs/tutorials/Martini3/ProteinsI/" and I found the issue that martinize2 -p backbone create NO restraints.

For example, I used "martinize2 -f 181L_clean.pdb -o t4l_only.top -x t4l_cg.pdb -dssp -p backbone" 
It return "martinize2 -f 181L_clean.pdb -o t4l_only.top -x t4l_cg.pdb -dssp -p backbone
```
    INFO - general - Read 1 molecules from PDB file 181L_clean.pdb
    INFO - step - Guessing the bonds.
    INFO - general - 1 molecules after guessing bonds
    INFO - step - Repairing the graph.
    INFO - general - Applying modification N-ter to residue A-MET1
    INFO - general - Applying modification C-ter to residue A-LYS162
    INFO - missing-atom - Missing atom LYS162:OXT
    INFO - step - Dealing with modifications.
    INFO - general - Identified the modifications ['N-ter'] on residues ['MET1', 'MET1', 'MET1', 'MET1']
    INFO - general - Identified the modifications ['C-ter'] on residues ['LYS162', 'LYS162', 'LYS162']
    INFO - step - Read input.
    INFO - step - Creating the graph at the target resolution.
    INFO - general - Applying modification mapping ('N-ter',)
    INFO - general - Applying modification mapping ('C-ter',)
    INFO - step - Averaging the coordinates.
    INFO - step - Applying the links.
    INFO - step - Placing the charge dummies.
    INFO - step - Applying position restraints.
    INFO - step - Sorting atomids
    INFO - step - Writing output.
    INFO - general - Please cite: Souza, P C T; Alessandri, R; Barnoud, J; Thallmair, S; Faustino, I; Grünewald, F; Patmanidis, I; Abdizadeh, H; Bruininks, B M H; Wassenaar, T A; Kroon, P C; Melcr, J; Nieto, V; Corradi, V; Khan, H M; Domański, J; Javanainen, M; Martinez-Seara, H; Reuter, N; Best, R B; Vattulainen, I; Monticelli, L; Periole, X; Tieleman, D P; de Vries, A H; Marrink, S J;  Nature Methods 2021; 10.1038/s41592-021-01098-3
    INFO - general - Please cite: McGibbon, R T; Beauchamp, K A; Harrigan, M P; Klein, C; Swails, J M; Hernández, C X; Schwantes, C R; Wang, L; Lane, T J; Pande, V S;  Biophysical Journal 2015; 10.1016/j.bpj.2015.08.015
    INFO - general - Backing up /mnt/d/mdmm/test/molecule_0.itp to /mnt/d/mdmm/test/#molecule_0.itp.1#.
    INFO - general - Backing up /mnt/d/mdmm/test/t4l_only.top to /mnt/d/mdmm/test/#t4l_only.top.1#.
    INFO - general - Backing up /mnt/d/mdmm/test/t4l_cg.pdb to /mnt/d/mdmm/test/#t4l_cg.pdb.1#.
    INFO - general - Shaken, not stirred. -- James Bond
```
i.e.,  one itp, one pdb, and one top file are created without the expected restraint files. It do not meet the information from "martinize2 -h" "Position restraints:
  -p {none,all,backbone}
                        Output position restraints (none/all/backbone) (default: none)
" 

How can I solve this issue?

**Comment by pckroon** [comment]
I tried `martinize2 -f 1ubq.pdb -p backbone -ignore HOH -x cg.pdb -o topol.top`, and `molecule_0.itp` contains the required `position_restraints` section, with the required `#ifdef POSRES`.

**Comment by Toolxx** [comment]
> I tried `martinize2 -f 1ubq.pdb -p backbone -ignore HOH -x cg.pdb -o topol.top`, and `molecule_0.itp` contains the required `position_restraints` section, with the required `#ifdef POSRES`.

Thanks for your help and reply. I think I got confused from the tutorial's instruction. So the restraint information is inside the "molecule_0.itp" file. So if I would like to produce a restraint.gro, it is out of scope of martinize2? Or how can I produce this based on the information from itp file? Thanks.

**Comment by pckroon** [comment]
I'm thinking you could just use the produced pdb file (`t4l_cg.pdb`), but I'm not sure whether gromacs will complain about the extra particles.
@csbrasnett do you know?

**Comment by csbrasnett** [comment]
Using the same file should be fine. @Toolxx just to expand on what I said over [on the discussion forum](https://github.com/orgs/Martini-Force-Field-Initiative/discussions/12): 

My understanding is that gromacs is happy as long as it finds enough coordinates in the right indices in the restraint file, there's no strict requirement for the restraint file to be precisely related to the coordinate input file.

So I have a coordinates.gro that looks like:
```
Title of the system in water
30152
    1MET     BB    1   7.016   9.231   7.402
    1MET    SC1    2   7.015   8.921   7.141
    2GLU     BB    3   7.255   9.464   7.321
    2GLU    SC1    4   7.533   9.288   7.550
    3SER     BB    5   7.277   9.516   6.978
    3SER    SC1    6   7.221   9.246   6.913
    4GLY     BB    7   7.587   9.427   6.859
    5SER     BB    8   7.653   9.158   7.058
...
```

and a topology file topol.top like: 

```
#include "martini_v3.0.0.itp"
#include "martini_v3.0.0_solvents_v1.itp"
#include "martini_v3.0.0_ions_v1.itp"
#include "molecule_0.itp"

[ system ]
Title of the system in water

[ molecules ]
molecule_0    1
W           28824
NA               336
CL               340
```

Then in `molecule_0.itp`, martinize2 has generated a `[ position_restraints ]` directive as discussed:

```
[ position_restraints ]
#ifdef POSRES
  1 1 1000 1000 1000
  3 1 1000 1000 1000
  5 1 1000 1000 1000
  7 1 1000 1000 1000
  8 1 1000 1000 1000
 10 1 1000 1000 1000
 12 1 1000 1000 1000
 14 1 1000 1000 1000
 16 1 1000 1000 1000
...
```

The minimization file that I have contains a `define` entry:

```
define = -DPOSRES -DFLEXIBLE
```

The position restraints matter at the `grompp` stage. What this means is when I run:

```
gmx grompp -f parameters.mdp -c coordinates.gro -p topol.top -o run.tpr
```

Gromacs will in this case produce a warning, because I have unused definitions in my system. This is corrected by:

```
gmx grompp -f parameters.mdp -c coordinates.gro -r coordinates.gro -p topol.top -o run.tpr
```

Which is useful in the case of e.g. energy minimization, because gromacs:
1) looks at parameters.mdp, and sees the `define = -DPOSRES`
2) looks through the molecules included in `topol.top`
3) sees the `#ifdef POSRES` section in `molecule_0.itp`
4) applies a position restraint as described to the atoms listed in the directive, to their respective positions in the `-r coordinates.gro`

To summarise, in most cases, the same coordinate file can (and should) be used for both the input coordinates and the position restraint coordinate file. Consider what happens if you *don't* do this. You're telling gromacs that you want to start a simulation from configuration `-c coordinates.gro`, but from the first integration step, you need to apply a harmonic force to restrain it to a completely different position defined in `-r restraint.gro`. The position restraints generated by martinize2 are assuming the first situation. If you're interested in other types of position restraints, then that's unfortunately not in the scope of the program.
